modules/images/webp-uploads/load.php

```diff
@@ -475,3 +475,43 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 }
 
 add_filter( 'the_content', 'webp_uploads_update_image_references', 10 );
+
+/**
+ * Updates the response for an attachment to include sources for additional mime types available the image.
+ *
+ * @since n.e.x.t
+ *
+ * @param WP_REST_Response $response The original response object.
+ * @param WP_Post          $post     The post object.
+ * @param WP_REST_Request  $request  The request object.
+ * @return WP_REST_Response A new response object for the attachment with additional sources.
+ */
+function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Post $post, WP_REST_Request $request ) {
+	$metadata = wp_get_attachment_metadata( $post->ID );
+	if ( empty( $metadata['file'] ) ) {
+		return $response;
+	}
+
+	$data = $response->get_data();
+	if ( ! isset( $data['media_details']['sizes'] ) || ! is_array( $data['media_details']['sizes'] ) ) {
+		return $response;
+	}
+
+	foreach ( $data['media_details']['sizes'] as $size => $details ) {
+		if ( empty( $metadata['sizes'][ $size ]['sources'] ) || ! is_array( $metadata['sizes'][ $size ]['sources'] ) ) {
+			continue;
+		}
+
+		$sources = array();
+		foreach ( $metadata['sizes'][ $size ]['sources'] as $mime => $mime_details ) {
+			$source_url       = dirname( $data['media_details']['sizes'][ $size ]['source_url'] ) . '/' . $mime_details['file'];
+			$sources[ $mime ] = array_merge( $mime_details, array( 'source_url' => $source_url ) );
```

Comment:
In addition to that avoiding the need to create a full new array traversing all the elements of the array just appending the new value seems good enough.
```suggestion
			$mime_details['source_url'] = $source_url;
			$sources[ $mime ] = $mime_details;
```

Reply:
Yep, also works! Thanks.